src/core/common/offset_range.hpp

```diff
@@ -0,0 +1,172 @@
+     * in the message up to its current length `aMessage.GetLength()`.
+     *
+     * @param[in] aMessage    The `Message` to initialize the `OffsetRange` from.
+     *
+     */
+    void InitFromMessageOffsetToEnd(const Message &aMessage);
+
+    /**
+     * Initializes the `OffsetRange` from a given `Message` from zero offset up to to its full length.
+     *
+     * The start offset of the range is set to zero, and the end offset is set to include full length of @p aMessage.
+     *
+     * @param[in] aMessage    The `Message` to initialize the `OffsetRange` from.
+     *
+     */
+    void InitFromMessageFullLength(const Message &aMessage);
+
+    /**
+     * Gets the start offset of the `OffsetRange`
+     *
+     * @returns The start offset.
+     *
+     */
+    uint16_t GetOffset(void) const { return mOffset; }
+
+    /**
+     * Gets the end offset of the `OffsetRange`.
+     *
+     * This offset is exclusive, meaning it marks the position immediately after the last byte within the range.
+     *
+     * @returns The end offset.
+     *
+     */
+    uint16_t GetEndOffset(void) const { return (mOffset + mLength); }
+
+    /**
+     * Gets the `OffsetRange` length.
+     *
+     * @returns The length of the `OffsetRange` in bytes.
+     *
+     */
+    uint16_t GetLength(void) const { return mLength; }
+
+    /**
+     * Indicates whether or not the `OffsetRange` is empty.
+     *
+     * @retval TRUE   The `OffsetRange` is empty.
+     * @retval FALSE  The `OffsetRange` is not empty (contains at least one byte).
+     *
+     */
+    bool IsEmpty(void) const { return (mLength == 0); }
+
+    /**
+     * Indicates whether or not the `OffsetRange` contains a given number of bytes.
+     *
+     * @param[in] aLength   The length to check.
+     *
+     * @retval TRUE   The `OffsetRange` contains @p aLength or more bytes.
+     * @retval FALSE  The `OffsetRange` does not contain @p aLength bytes.
+     *
+     */
+    bool Contains(uint32_t aLength) const { return aLength <= mLength; }
+
+    /**
+     * Advances the start offset forward by the given number of bytes.
+     *
+     * This method ensures the start offset does not go beyond the end offset of the `OffsetRange`. If @p aLength is
+     * greater than the available bytes in the `OffsetRange`, the start offset is adjusted to the end offset,
+     * effectively shrinking the range to zero length.
+     *
+     * @param[in]  aLength   The number of bytes to advance the start offset.
+     *
+     */
+    void AdvanceOffset(uint32_t aLength);
+
+    /**
+     * Adjusts the `OffsetRange` length to a given length, if its current length is longer.
+     *
+     * If the current length of the `OffsetRange` is longer than @p aLength, the offset range is shortened to
+     * @p aLength. If its current length is shorter or the same, the `OffsetRange` remains unchanged.
+     *
+     * @param[in] aLength  The new length to use.
+     *
+     */
+    void AdjustLength(uint16_t aLength);
```

Comment:
What about naming it as `SetIfLonger`? `AdjustLength` works for me as well.

Reply:
Thanks for the review.

I renamed the method from `AdjustLength()` to `ShrinkLength()`. I think it is a better name and and makes it clear that we can only make the range smaller.

Reply:
Sounds good!